lib/bookshelf-slug.js

```diff
@@ -70,46 +71,36 @@ module.exports = function(bookshelf, settings) {
     },
 
     __beginSlug: function(value) {
-      return Promise.resolve()
-        .then(_ => {
-          return this.__checkSlug(value)
-            .then(isUnique => {
-                if(isUnique) {
-                  return this.set(this.__slug.column, value);
-                }
-                return this.__incrementSlug(value);
-            });
-        })
+      return this.__checkSlug(value)
+        .then(isUnique => {
+            if(isUnique) {
+              return this.set(this.__slug.column, value);
+            }
+            return this.__incrementSlug(value);
+        });
     },
 
     __incrementSlug: function(value) {
       var inc = (newSlug, count) => {
         newSlug = newSlug + '-' + Date.now() +'-'+ count;
-        return Promise.resolve()
-          .then(_ => {
-            return this.__checkSlug(newSlug, this.__slug.column)
-              .then(isUnique => {
-                if(isUnique) {
-                  return this.set(this.__slug.column, newSlug);
-                }
-                return inc(newSlug, count + 1);
-              })
-          })
+        return this.__checkSlug(newSlug, this.__slug.column)
```

Comment:
This seems to not be working as expected, I need to verify this part.

Reply:
Yes that was a bug, but the bug was related to `model.hasChanged` and to be able to detect that the field has changed to prevent the __increment codepath.

Reply:
Did you manage to fix this bug?

Reply:
Yes it has been resolved and I think the tests I added covers this as well.